helper/resource/testing.go

```diff
@@ -570,6 +573,19 @@ type TestStep struct {
 	// at the end of the test step that is verifying import behavior.
 	ImportStatePersist bool
 
+	//---------------------------------------------------------------
+	// RefreshState testing
+	//---------------------------------------------------------------
+
+	// RefreshState, if true, will test the functionality of `terraform
+	// refresh` by refreshing the state.
+	RefreshState bool
+
+	// RefreshStateCheck checks state following `terraform refresh`. It
+	// should be used to verify that the state has the expected resources,
+	// IDs, and attributes.
+	RefreshStateCheck RefreshStateCheckFunc
```

Comment:
For the use case I'm picturing, I'd prefer to be able to reuse the existing ecosystem of `TestCheckFunc`s after the refresh. This would allow tests to run against the whole refreshed state.